allennlp/training/optimizers.py

```diff
@@ -121,7 +121,14 @@ def from_params(cls, model_parameters: List, params: Params):  # type: ignore
             else:
                 num_parameters += parameter_group.numel()
         logger.info("Number of trainable parameters: %s", num_parameters)
-        return Optimizer.by_name(optimizer)(parameter_groups, **params.as_dict()) # type: ignore
+
+        # By default we cast things that e.g. look like floats to floats before handing them
+        # to the Optimizer constructor, but if you want to disable that behavior you could add a
+        #       "infer_type_and_cast": false
+        # key to your "trainer" config.
```

Comment:
Nit: "your "trainer.optimizer" config"?.  Also, it'd be nice if this were discoverable somehow, but that's a more general problem that I'm not sure how to solve...